contracts/RMRK/extension/erc20Holder/ERC20Holder.sol

```diff
@@ -0,0 +1,127 @@
+// SPDX-License-Identifier: Apache-2.0
+
+pragma solidity ^0.8.18;
+
+import "./IERC20Holder.sol";
+import "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+
+error InvalidValue();
+error InvalidAddress();
+error InsufficientBalance();
+
+abstract contract ERC20Holder is IERC20Holder {
+    mapping(uint256 => mapping(address => uint256)) private balances;
+
+    /**
+     * @inheritdoc IERC20Holder
+     */
+    function balanceOfERC20(
+        address erc20Contract,
+        uint256 tokenId
+    ) external view returns (uint256) {
+        return balances[tokenId][erc20Contract];
+    }
+
+    /**
+     * @inheritdoc IERC20Holder
+     */
+    function transferERC20FromToken(
+        address erc20Contract,
+        uint256 tokenId,
+        address to,
+        uint256 value
```

Comment:
I'm thinking of adding a `bytes data` here, for advanced usage. Same on the transferTo method.

Reply:
Yeah, I think it makes sense to include it 👌